source/host_resolver.c

```diff
@@ -1360,15 +1361,17 @@ static inline int create_and_init_host_entry(
     struct aws_thread_options thread_options = *aws_default_thread_options();
     thread_options.join_strategy = AWS_TJS_MANAGED;
 
-    aws_thread_launch(&new_host_entry->resolver_thread, resolver_thread_fn, new_host_entry, &thread_options);
+    aws_thread_launch(&resolver_thread, resolver_thread_fn, new_host_entry, &thread_options);
```

Comment:
Thank you for your PR.

> You must either call join or detach after creating the thread and before calling clean_up.

Referring to [here](https://github.com/awslabs/aws-c-common/blob/main/include/aws/common/thread.h#L119). I think we cannot just make the thread a local variable and clean it up like that. But seems like we do forget about thread clean up when clean up the host_entry. Not super sure, still investigating.

Reply:
Maybe instead of making it a local variable, add `aws_thread_clean_up(&entry->resolver_thread)` to `s_clean_up_host_entry`?

Reply:
It seemed to me that when creating a managed thread calling join and clean up will be taken care of by the thread wrapper...